tests/cypress/e2e/helpers.ts

```diff
@@ -55,14 +55,45 @@ export function deleteCmkHost(hostName: string) {
   });
 }
 
+function waitForActivation(activation_id: string, waitingTime: number = 1000) {
+  cy.wait(waitingTime);
+  cy.request({
+    method: 'GET',
+    url:
+      Cypress.env('cypressToCheckmkUrl') +
+      '/check_mk/api/1.0/objects/activation_run/' +
+      activation_id +
+      '/actions/wait-for-completion/invoke',
+    followRedirect: false,
+    headers: { accept: 'application/json' },
+    auth: {
+      bearer: `${Cypress.env('cmkUsername')} ${Cypress.env('cmkPassword')}`,
+    },
+    body: {
+      redirect: false,
+    },
+  }).then((response) => {
+    if (response.status === 204) return;
+
+    waitForActivation(activation_id);
+  });
+}
+
 export function activateCmkChanges(siteName: string) {
+  Cypress.on('uncaught:exception', (err) => {
```

Comment:
are you sure this fixes `Could not find data source` of https://github.com/tribe29/grafana-checkmk-datasource/actions/runs/3722750224/jobs/6313829415 ? 
`Could not find data source` is a grafana error. do you know when/why this error occurs? could there be another solution? it feels strange that we catch a grafana error here in the checkmk code.
my guess would be, that this is an error from the previous check (configuring the data source). can you investigate here?
this looks a bit too general for me. if possible we should find the root cause and fix it there.

Reply:
Discussed f2f.